lib/shared/src/models/sync.ts

```diff
@@ -323,7 +335,74 @@ export function syncModels({
                                                 ) {
                                                     data.preferences!.defaults.chat = haikuModel.id
                                                 }
+                                                if (authStatus.rateLimited) {
+                                                    // Disable all the non-fast models
+                                                    data.primaryModels = data.primaryModels.map(
+                                                        model => {
+                                                            if (!model.tags.includes(ModelTag.Speed)) {
+                                                                return { ...model, disabled: true }
+                                                            }
+                                                            return model
+                                                        }
+                                                    )
 
+                                                    // Set Gemini Flash as the selected model
+                                                    const geminiFlashModel = data.primaryModels.find(
+                                                        model =>
+                                                            model.id === 'google::v1::gemini-2.0-flash'
+                                                    )
+                                                    if (geminiFlashModel && data.preferences) {
+                                                        rateLimitState.chatModel =
+                                                            data.preferences.defaults.chat || ''
+                                                        rateLimitState.editModel =
+                                                            data.preferences.defaults.edit || ''
+
+                                                        data.preferences.defaults.chat =
+                                                            geminiFlashModel.id
+                                                        data.preferences.defaults.edit =
+                                                            geminiFlashModel.id
+                                                        telemetryRecorder.recordEvent(
+                                                            'cody.rateLimit',
+                                                            'hit',
+                                                            {
+                                                                privateMetadata: {
+                                                                    rateLimitedModel:
```

Comment:
Could we update this to `chatModel`? We allow listed this field since it is not sensitive and by naming it `chatModel` we would then be able to collect this information for our enterprise users as well.

```suggestion
                                                                    chatModel:
```

Reply:
non-blocking comment/suggestion. This can be done at a later time in a different PR :)